rsconnect/actions.py

```diff
@@ -481,7 +481,9 @@ def deploy_jupyter_notebook(
     (app_id, deployment_name, deployment_title, default_title, app_mode,) = gather_basic_deployment_info_for_notebook(
         connect_server, app_store, file_name, new, app_id, title, static
     )
-    python, environment = get_python_env_info(file_name, python, compatibility_mode, force_generate)
+    python, environment = get_python_env_info(
+        file_name, python, conda_mode=not compatibility_mode, force_generate=force_generate,
```

Comment:
I get the rationale behind not altering the arg name in the func signature, although I feel like here you have a workable argument that a mismatching `rsconnect-jupyter` version is absolutely unsupported and crashing on a syntax error may actually count as more graceful degradation than surreptitiously capturing a user's conda environment.

Reply:
Is your recommendation that `compatibility_mode` be changed to `conda_mode` throughout this project?

Reply:
Yes, although I think doing so could be a separate issue due to the potential fallout on the jupyter plugin from calling the function incorrectly.